core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -25,32 +25,84 @@ oppia.constant(
 oppia.constant(
   'COLLECTION_RIGHTS_URL_TEMPLATE',
   '/collection_editor_handler/rights/<collection_id>');
+oppia.constant(
+  'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
+  '/explorationsummarieshandler/data?' +
+  'stringified_exp_ids=<stringified_exp_ids>&' +
+  'include_private_explorations=<include_private_explorations>');
 
 oppia.controller('CollectionEditor', ['$scope',
   'WritableCollectionBackendApiService', 'CollectionRightsBackendApiService',
-  'CollectionObjectFactory', 'SkillListObjectFactory',
+  'ExplorationSummaryBackendApiService', 'CollectionObjectFactory',
+  'SkillListObjectFactory', 'CollectionValidationService',
   'CollectionUpdateService', 'UndoRedoService', 'alertsService', function(
     $scope, WritableCollectionBackendApiService,
-    CollectionRightsBackendApiService, CollectionObjectFactory,
-    SkillListObjectFactory, CollectionUpdateService, UndoRedoService,
+    CollectionRightsBackendApiService, ExplorationSummaryBackendApiService,
+    CollectionObjectFactory, SkillListObjectFactory,
+    CollectionValidationService, CollectionUpdateService, UndoRedoService,
     alertsService) {
     $scope.collection = null;
     $scope.collectionId = GLOBALS.collectionId;
     $scope.collectionSkillList = SkillListObjectFactory.create([]);
     $scope.isPublic = GLOBALS.isPublic;
+    $scope.validationIssues = [];
+
+    var _validateCollection = function() {
+      if ($scope.isPublic) {
```

Comment:
Just to confirm: we don't have the concept of a 'publicized' ('featured') collection, right? I ask because, if so, it may be more prudent to use "not private" here instead (and maybe even change the globals property to isPrivate).

There's an unfortunate terminology issue here: for explorations, a status of "public" in the rights model means "not private but also not featured". I suspect that all you care about here is "not private".